sdks/python/apache_beam/runners/portability/fn_api_runner/execution.py

```diff
@@ -227,12 +251,24 @@ def __iter__(self):
     """
     return itertools.chain(*self.partition(1))
 
+  # these should never be accessed, but they allow this class to meet the
```

Comment:
How about making them raise errors rather than be no-ops?

Reply:
I prefer the current solution because it's reasonable and safe to call `clear()` and `reset()` on this class and have it do nothing:  there's simply nothing to clear or reset.   It seems wrong to meet the protocol definition, but then error if the methods are called: that's not really meeting the protocol.